huey/backends/redis_backend.py

```diff
@@ -78,12 +78,25 @@ def read(self):
             return None
 
 
+# a custom lua script to pass to redis that will read tasks from the schedule
+# and atomically pop them from the sorted set and return them.
+# it won't return anything if it isn't able to remove the items it reads.
+SCHEDULE_POP_LUA = """
+local key = KEYS[1]
+local unix_ts = ARGV[1]
+local res = redis.call('zrangebyscore', key, '-inf', unix_ts)
+if #res and redis.call('zremrangebyscore', key, '-inf', unix_ts) == #res then
+    return res
+end
```

Comment:
FYI, I've been using this patch on my dev server for the past week and everything is working flawlessly.